persistent-sqlite/Database/Persist/Sqlite.hs

```diff
@@ -64,7 +64,12 @@ open' connStr logFunc = do
             | Just cs <- T.stripPrefix "WAL=off " connStr -> (cs, False)
             | otherwise                                   -> (connStr, True)
 
-    conn <- Sqlite.open connStr'
+    let (connStr'', readOnly) = case () of
+          ()
+            | Just cs <- T.stripPrefix "READONLY " connStr'  -> (cs, True)
+            | Just cs <- T.stripPrefix "READWRITE " connStr' -> (cs, False)
+            | otherwise                                      -> (connStr', False)
+    conn <- Sqlite.open connStr'' readOnly
```

Comment:
One thought is that, long-term, this function should probably move away from passing and parsing a connection string. If there was a `SqliteOpts` data structure with default values (or something like that), this code would be simpler and more scalable for supporting future options. The main downside is that you'd have to duplicate the existing APIs that accept a connection string. Thoughts @gregwebs ?